### PR TITLE
Created developer tools.

### PR DESCRIPTION
**Pull request by diabolical17**
More can be added to this, such as access keys etc. But thats really for after beta.

*Pressing Continue on the revoke token screen simply logs the user out and redirects to the homepage*

Gif
Gif

### CODE REVIEW

**Comment by Geometrically** [comment]
Could you move this to the settings page instead of a separate page?

**Comment by diabolical17** [comment]
Yeah sure. Should i move all the settings related stuff to a central settings page? A bit like GitHub?

File: pages/dashboard/developer.vue [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+            >Beware, this will log you out of Modrinth, and you will have to
+            login again to access modrinth with a new token.</span
+          >
+          <button @click="showDelPopup">Revoke Token</button>
+        </label>
+      </section>
+    </modal>
+    <modal v-show="revoke_component">
+      <RevokeTokenHint @wantClose="showDelPopup"></RevokeTokenHint>
+    </modal>
```
Comment (Geometrically) [inline pages/dashboard/developer.vue:28]: Yeah, should be just moving this and all related code to the `/dashboard/settings`

**Comment by diabolical17** [comment]
Why not just account/settings or settings/...?

**Comment by diabolical17** [comment]
Why not just account/settings or settings/...?

**Comment by Geometrically** [comment]
Because it's part of the dashboard

**Comment by diabolical17** [comment]
@Geometrically See updated gifs on the PR description for changes

**Comment by diabolical17** [comment]
Alright, this should be ready.

File: assets/images/sidebar/devtools.svg [hunk 2]
```diff
@@ -0,0 +1 @@
+<svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-tool"><path d="M14.7 6.3a1 1 0 0 0 0 1.4l1.6 1.6a1 1 0 0 0 1.4 0l3.77-3.77a6 6 0 0 1-7.94 7.94l-6.91 6.91a2.12 2.12 0 0 1-3-3l6.91-6.91a6 6 0 0 1 7.94-7.94l-3.76 3.76z"></path></svg>
```
Comment (Geometrically) [inline assets/images/sidebar/devtools.svg:1]: Where did you get this from? Does this have a license compatible with AGPL?

File: pages/dashboard/settings.vue [hunk 3]
```diff
@@ -4,69 +4,115 @@
...
+          <button type="button" class="button" @click="showDelPopup">
+            Revoke token
+          </button>
+        </label>
+      </section>
+    </modal>
+    <modal v-show="revoke_component">
+      <RevokeTokenHint @wantClose="showDelPopup"></RevokeTokenHint>
+    </modal>
   </DashboardPage>
```
Comment (Geometrically) [inline pages/dashboard/settings.vue:71]: This seems like a bad way to this.. Why not just have the revoke hint be on a separate page?

File: pages/dashboard/settings.vue [hunk 4]
```diff
@@ -123,5 +185,15 @@ label {
...
+  button {
+    :hover {
+      cursor: pointer;
+    }
+    height: fit-content;
+    flex: 1;
+  }
+}
+.header-top {
+  font-size: 125%;
```
Comment (Geometrically) [inline pages/dashboard/settings.vue:197]: Why is this needed?

**Review by Geometrically — CHANGES_REQUESTED** [review]

File: assets/images/sidebar/devtools.svg (hunk 2 above)
Comment (diabolical17) [reply assets/images/sidebar/devtools.svg:1]: The trello?

File: pages/dashboard/settings.vue (hunk 4 above)
Comment (diabolical17) [reply pages/dashboard/settings.vue:197]: Its actually not used. So i'll delete it.

File: assets/images/sidebar/devtools.svg (hunk 2 above)
Comment (Geometrically) [reply assets/images/sidebar/devtools.svg:1]: We have a trello?

Comment (diabolical17) [reply assets/images/sidebar/devtools.svg:1]: https://trello.com/modrinth

File: pages/dashboard/settings.vue (hunk 3 above)
Comment (diabolical17) [reply pages/dashboard/settings.vue:71]: Whats the point of it being on another page? Why make the user navigate to a seperate page just to give them 4 sentences and a button that logs them out?

Comment (diabolical17) [reply pages/dashboard/settings.vue:71]: Also, this modal system can be extended in the future. I'll be making another PR in the future which will wrap modals allowing for multiple sections. EG: A wizard for modpack creation...

**Comment by diabolical17** [comment]
I'll rework the "Are you sure you want to revoke your token page." And try make it a seperate page

**Comment by diabolical17** [comment]
Closing for now, will make a new PR when ready.
